### PR TITLE
#502 Add SchemaVersion field into PKI, Compliance, Model and VendorInfo schemas

### PR DESCRIPTION
**Pull request by Abdulbois**

### CODE REVIEW

File: docs/transactions.md [hunk 1]
```diff
@@ -128,6 +128,7 @@ Adds a record about a Vendor.
   - companyLegalName: `string` -  Legal name of the vendor company
   - companyPreferredName: `optional(string)` -  Preferred name of the vendor company
   - vendorLandingPageURL: `optional(string)` -  URL of the vendor's landing page
```
Comment (ashcherbakov) [inline docs/transactions.md:130]: Please make sure that we have schemaVersion in docs for the following commands:
- ADD_PKI_REVOCATION_DISTRIBUTION_POINT
- UPDATE_PKI_REVOCATION_DISTRIBUTION_POINT
- Approve/Reject/Revoke wrappers for PKI commands (see https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/552#discussion_r1533681328)

**Review by ashcherbakov — CHANGES_REQUESTED** [review]
- I think we need to have schemaVersion for auxiliary models as well: Accounts, Validator, Update.
- Please don't forget to update the PR description

**Comment by akarabashov** [comment]
> * I think we need to have schemaVersion for auxiliary models as well: Accounts, Validator, Update.

Created task: https://github.com/zigbee-alliance/distributed-compliance-ledger/issues/556

File: proto/compliance/compliance_info.proto [hunk 2]
```diff
@@ -30,6 +30,6 @@ message ComplianceInfo {
   string OSVersion = 21;
   string parentChild = 22;
   string certificationIdOfSoftwareComponent = 23;
-  
+  uint32 schemaVersion = 24;
```
[the remark this answers is not in the document]
Comment (Abdulbois) [reply proto/compliance/compliance_info.proto:33]: Concern to make it `uint32`was that several existing Version fields in our schemas are `uint32`
Can we leave it as `uint32` or making it `uint16` strictly required?

**Comment by Abdulbois** [comment]
> As the new field is optional, and we have to manually pass it to the object constructor in every handler and every CLI command, there is a risk that we may unintentionally forget to pass it somewhere. So, I would recommend to add such tests for **every** command. Either have a dedicated/separate test where we explicitly set schema version for all add/update commands and check that it has been set/updated, or extend existing Add/Update tests. I know that we already have such tests for some transactions, I think it's important to have it for all of them.
> 
> **Note**: it seems such checks (that we pass and apply optional parameters correctly) may be duplicated in unit and CLI tests. Although usually we prefer unit tests, in this case we may consider having CLI tests only (as it implicitly covers the unit test cases as well).

I think, currently all txn CLI calls are checked for using `schemaVersion` in their commands
To be sure wiil double check

File: proto/compliance/compliance_info.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply proto/compliance/compliance_info.proto:33]: unit16 is defined by the Spec, our main document...
Please note though, that in the past we experienced some difficulties with uint16 on the .proto level, so it's defined as int32/unit32 in the main .proto, and restrictions are defined in `tx.proto`, see for example `vid` and `pid` fields.

[the remark this answers is not in the document]
Comment (Abdulbois) [reply proto/compliance/compliance_info.proto:33]: Fixed with adding validation tag into `tx.proto` messages

**Review by ashcherbakov — APPROVED** [review]

File: docs/transactions.md [hunk 3]
```diff
@@ -128,6 +128,7 @@ Adds a record about a Vendor.
...
+  - schemaVersion: `optional(uint32)` - Schema version to support backward/forward compatability(default 0)
```
Comment (akarabashov) [inline docs/transactions.md:131]:
```suggestion
  - schemaVersion: `optional(uint16)` - Schema version to support backward/forward compatability(default 0)
```
Please correct it in other places as well

**Review by akarabashov — COMMENTED** [review]
I noticed that unit tests for static validation are missing, I think it makes sense to add them

**Comment by Abdulbois** [comment]
> I noticed that unit tests for static validation are missing, I think it makes sense to add them

Added, please recheck

**Review by akarabashov — APPROVED** [review]

**Pull request merged**